### PR TITLE
Handle equal sign in Extra Args

### PR DESCRIPTION
**Pull request by DanielXiao**
Various EXTRA_ARGS are required of format "key1=value1;key2=value2", and some of arg value also contains equal sign, for example: feature-gates=RotateKubeletServerCertificate=true,QOSReserved=true We should take everything after the the first equal sign as its value in this case.

Move converting legacy variables to Cluster variables to the common method get_cluster_variables() because there is no difference among providers.

Validate legacy variables on various EXTRA_ARGS.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: tkg/client/cluster.go [hunk 1]
```diff
@@ -738,13 +738,46 @@ func (c *TkgClient) ConfigureAndValidateWorkloadClusterConfiguration(options *Cr
...
+	}
+
 	if err = c.ValidateKubeVipLBConfiguration(TkgLabelClusterRoleWorkload); err != nil {
 		return NewValidationError(ValidationErrorCode, err.Error())
 	}
 
 	return c.ValidateSupportOfK8sVersionForManagmentCluster(clusterClient, options.KubernetesVersion, skipValidation)
 }
 
+func (c *TkgClient) ConfigAndValidateExtraArgs() error {
```
Comment (hongtaiwang) [inline tkg/client/cluster.go:752]: seems we only do validation for ExtraArgs in this func, what about renaming the func to avoid confusion?

Comment (DanielXiao) [reply tkg/client/cluster.go:752]: We also set empty string if not found

**Review by hongtaiwang — APPROVED** [review]
lgtm

**Comment by zhouke1991** [comment]
LGTM!

**Review by wjun — APPROVED** [review]

**Comment by alfredthenarwhal** [comment]
Created cherry-pick PR

**Pull request merged**